src/assignments/tech-jobs-console.rst

```diff
@@ -0,0 +1,511 @@
+   after time. To kill it, press the red “stop” icon in the Run pane. We’ll
+   learn precisely how the program manages to work this way below.
+
+Let’s explore the code by starting with the source of the data our program is
+providing access to.
+
+The Data File: ``jobs_data.csv``
+---------------------------------
+
+Our simple app doesn’t connect to a database. If the prototype proves
+useful and we continue development, we’ll add that functionality later.
+But for now, we’ve been given a CSV (comma-separated values) file from
+the Company Team at LaunchCode that contains some recent jobs. This file
+was exported from an Excel spreadsheet into this format, which is easy
+for programs to read in.
+
+If CSV files are new to you, don’t worry, they’re easy to understand.
+CSV files are conceptually similar to simple spreadsheets in that they
+organize data in rows and columns. The major difference is that they
+don’t have the ability to carry out calculations the way spreadsheets
+do, and you can easily open, read, and edit them in plain text editors.
+
+Open up ``jobs_data.csv``, which is in the ``resources`` folder at the
+top level of the project. You’ll see that the first line is:
+
+.. sourcecode:: bash
+
+   name,employer,location,position type,core competency
+
+While it isn’t required, the first line of a CSV file often represents
+the column names. We have 5 names here, which indicates that each of our
+rows in the CSV file should have 5 fields. In this file format, a “row”
+corresponds to a new line. So each line below the first will constitute
+a row of data, or a record.
+
+Have a look a the data below line 1, and ask yourself the following
+questions: - Which fields match up with which column names above? - Why
+do some lines/rows (e.g. line 10) have more commas than others, if
+commas are supposed to separate columns? - What role do the
+double-quotes play?
+
+The TechJobs Class
+-------------------
+
+The ``TechJobs`` class contains the ``main`` method that will drive our
+program’s functionality. It contains three methods:
+
+#. ``main`` - The main application runner.
+#. ``getUserSelection`` - A utility method that displays a menu of choices and
+   returns the user’s selection.
+#. ``printJobs`` - This is meant to print a list of jobs to the console in a
+   nicely formatted manner, but hasn’t been implemented yet. This will be part
+   of your job.
+
+Let’s look at each of these.
+
+The ``main`` Method
+^^^^^^^^^^^^^^^^^^^^
+
+The logic within ``main`` presents menus in turn, and based on the
+user’s choice, takes appropriate action.
+
+It begins by declaring two local variables: ``columnChoices`` and
+``actionChoices``. These contain information relating to the menus that
+we’ll display, and we’ll look at them in more detail later.
+
+Next, we notice a ``while loop`` that starts ``while (true)``. This may
+seem odd, but actually, it makes a lot of sense after a short
```

Comment:
The sentence "This may seem odd..." is awkward. Perhaps, "While this may seem counterintuitive, we have a good reason for doing so!"

Reply:
Fixed.

Reply:
Looks good!